lib/src/update.dart

```diff
@@ -19,48 +19,65 @@ import 'package:yaml/yaml.dart';
 import 'package:abide/src/constants.dart';
 import 'package:abide/src/util.dart';
 
-Future<Null> updateAnalysisOption(YamlMap abideYaml,
-    {bool uncommentClean}) async {
-  final YamlMap currentAnalysisOptions =
-      loadAnalysisOptions(renameDeprecatedFilename: true);
-  final String currentAnalysisOptionsString = loadAnalysisOptionsAsString();
+Future<String> updateAnalysisOption(YamlMap abideYaml,
```

Comment:
Changed the return type to return the analysis options file contents as a String. This aids in testing since you don't actually have to do filesystem operations. Before it would assume it would read and write the file to the current directory. While that was convenient, it made running tests difficult. Changing the working directory changes it for the whole process. That meant tests couldn't be run concurrently. Instead, I changed the functions to return the content and have a flag on whether or not to actually write to the filesystem and the path to the file to read from.